### PR TITLE
 Fix current_user method & add custom header option

### PR DESCRIPTION
**Pull request by max-konin**
Hi 

I fixed current_user method after migrate to knock 2.x and added specs. 
Also I added option for header name. It can be useful

### CODE REVIEW

File: grape-knock.gemspec [hunk 1]
```diff
@@ -21,8 +21,10 @@ Gem::Specification.new do |spec|
 
   spec.add_dependency "grape", ">= 0.10"
   spec.add_dependency "knock"
+  spec.add_dependency "rails", ">= 4.2"
```
Comment (gottfrois) [inline grape-knock.gemspec]: why the deps on rails?

Comment (max-konin) [reply grape-knock.gemspec]: Knock is a Rails engine. You can not use it without rails... Without it I have error "Rails constant not found" when I run `rspec spec`

Comment (max-konin) [reply grape-knock.gemspec]: @gottfrois What do you think about it?

Comment (gottfrois) [reply grape-knock.gemspec]: i'm suprised we need the explicit dependency here since it's already declare in `knock` gem

Comment (max-konin) [reply grape-knock.gemspec]: Strangely.... I removed dependency and specs passed.... mb it is my mistake

**Comment by gottfrois** [comment]
Looks good! Thx :)

**Pull request merged**